test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -231,6 +234,13 @@ void test_transform_effect(VecIt1 first1, VecIt1 last1, VecIt2 first2) {
         }
 }
 
+struct transform_functor {
```

Comment:
How is it different from `test_lambda` below? Why do you need that?

Reply:
`test_lambda` is not default constructible, so we can not check if `transform_iterator` with `test_lambda` is default constructible.